### DISCUSSION TITLE
Add sorting based on file types, combine it with coloring

### DISCUSSION THREAD

**Discussion by ovdikklop** [doublecmd/doublecmd#1132] [Suggestions]
### Description

Add to "File views -> sort methods" somewhat like "Put listed in Colors->File types first", add to "Colors->File types" buttons to reorder types. Or even merge these settings together.
Here is how this done in far manager - you can adjust custom ordering and higlighting from one place
![изображение](https://github.com/doublecmd/doublecmd/assets/112009726/41c3012e-7d33-47d7-8099-5d213269f3fa)
It just handy to have an option to put, say, executables or something another depending on wildcard or regexp before or after anything else regardless of sort mode. To turn this feature on or off it would be good to add something like trigger in menu "Show->Use sorting groups" or alike just after Reverse order and Refresh.

**Comment by Skif-off** [comment]
I.e. sorting by group of files?
Maybe just additional column with suitable WDX plugin? [Groups](http://totalcmd.net/plugring/groups2.html) or something like this (cross-platform)
```lua
-- groupswdx.lua (cross-platform)
--2023.07.16
--[[
Returns the group name by the file mask
(see/edit the "aGroups" table below).
]]

local aGroups = {
{"Report",   "report_*"},
{"Executable (Win)", "*.bat;*.cmd;*.com;*.exe;*.js;*.msi;*.msu;*.vbs;*.wsf"},
{"Document", "*.chm;*.djvu;*.doc;*.docx;*.epub;*.odp;*.ods;*.odt;*.pdf;*.ppt;*.pptx;*.xls;*.xlsx"},
{"Image",    "*.avif;*.bmp;*.gif;*.heic;*.heif;*.jpg;*.jpeg;*.png;*.psd;*.tif;*.tiff;*.webp"},
{"Video",    "*.avi;*.mpg;*.mp4;*.mov;*.mkv;*.wmv"}
}

function ContentGetSupportedField(FieldIndex)
  if FieldIndex == 0 then
    return "Group", "", 8; -- FieldName, Units, ft_string
  end
  return "", "", 0; -- ft_nomorefields
end

function ContentGetDefaultSortOrder(FieldIndex)
  return 1
end

function ContentGetDetectString()
  return 'EXT="*"'; -- return detect string
end

function ContentGetValue(FileName, FieldIndex, UnitIndex, flags)
  if FieldIndex == 0 then
    local bResult, sResult
    local sName = SysUtils.ExtractFileName(FileName)
    local iCount = 0
    while true do
      iCount = iCount + 1
      bResult = SysUtils.MatchesMaskList(sName, aGroups[iCount][2], ";", 0)
      if bResult == true then
        sResult = aGroups[iCount][1]
        break
      end
      if iCount == #aGroups then break end
    end
    return sResult
  end
  return nil
end
```
?

P.S. Alse see [sort by multiple columns](https://doublecmd.github.io/doc/en/help.html#iface_columns).
